packages/userscript/source/WorkshopManager.ts

```diff
@@ -289,19 +290,33 @@ export class WorkshopManager extends UpgradeManager {
         // since the regulation happens there
         delta = this.getValueAvailable(resource) / materialAmount;
       } else {
-        // Take the currently present amount of material to craft into account
-        // Currently this determines the amount of resources that can be crafted such that base
-        // materials are proportionally distributed across limited resources.
-        // This base material distribution is governed by limRat "limited ratio" which defaults
-        // to 0.5, corresponding to half of the possible components being further crafted.
-        // If this were another value, such as 0.75, then if you had 10000 beams and 0 scaffolds,
-        // 7500 of the beams would be crafted into scaffolds.
-        delta =
-          limRat *
-            ((this.getValueAvailable(resource, true) +
-              (materialAmount / (1 + ratio)) * this.getValueAvailable(res.name, true)) /
-              materialAmount) -
-          this.getValueAvailable(res.name, true) / (1 + ratio);
+        // Quantity of source and target resource currently available.
+        const src_available = this.getValueAvailable(resource, true);
+        const tgt_available = this.getValueAvailable(name, true);
+
+        // How much source resource is consumed and target resource is crafted per craft operation.
+        const recipe_requires = materialAmount;
+        const recipe_produces = 1 + ratio;
+
+        // How many crafts could we do given the amount of source resource available.
+        const crafts_possible = src_available / recipe_requires;
+
+        // How many crafts were hypothetically done to produce the current amount of target resource.
+        const crafts_done = tgt_available / recipe_produces;
+
+        // Craft only when the crafts_possible >= crafts_done.
+        // Crafting gets progressively more expensive as the amount of the target increases.
+        // This heuristic gives other, cheaper, targets a chance to get built from the same source resource.
+        // There is no checking if there actually exists a different target that could get built.
+        delta = 1 + crafts_possible - crafts_done;
+
+        // If crafting is not going to happen, explain why not.
+        const explanation_messages = true;
```

Comment:
I'm not opposed to adding analytical messages to the script. But this local solution doesn't seem like a great fit to me.

What do you think about adding a property like `verbose`, `debug` or `analytics` to the `Options`? I don't think we'd need a UI element for this, as users could just toggle it with `kittenScientists.options.verbose = true` on the JS console.

Also, `cdebug` might be more appropriate for this message.